lib/SILOptimizer/Mandatory/DiagnoseVarUsage.cpp

```diff
@@ -0,0 +1,1617 @@
+//===-------- DiagnoseVarUsage.cpp - Perform var usage diagnostics --------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements a diagnostic pass to diagnose unused, never read, and
+// never mutated values.
+//
+//===----------------------------------------------------------------------===//
+
+#define DEBUG_TYPE "diagnose-var-usage"
```

Comment:
Where is this macro used?

Reply:
This macro is used within the `LLVM_DEBUG` macro in `DiagnoseVarUsage::checkVarUsage`